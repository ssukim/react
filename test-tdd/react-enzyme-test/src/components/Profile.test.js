import React from "react";
import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("스냅샷 매칭", () => {
    const wrapper = mount(<Profile username="ssukim" name="김정수" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("username & name 랜더링", () => {
    const wrapper = mount(<Profile username="ssukim" name="김정수" />);
    expect(wrapper.props().username).toBe("ssukim");
    expect(wrapper.props().name).toBe("김정수");

    const boldElement = wrapper.find('b');
    expect(boldElement.contains('ssukim')).toBe(true);
    const spanElement = wrapper.find('span');
    expect(spanElement.text()).toBe('(김정수)');
  });
});
