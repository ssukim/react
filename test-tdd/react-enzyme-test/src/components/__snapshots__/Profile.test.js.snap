// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> 스냅샷 매칭 1`] = `
<Profile
  name="김정수"
  username="ssukim"
>
  <div>
    <b>
      ssukim
    </b>
     
    <span>
      (
      김정수
      )
    </span>
  </div>
</Profile>
`;
